docs/design/Non_Admin_Controller_design.md

```diff
@@ -0,0 +1,83 @@
+# Non-Admin Backup/Restore Design
+
+## Background
+OADP (Openshift API for Data Protection) Operator currently requires cluster admin access for performing Backup and Restore operations of applications deployed on the OpenShift platform. This design intends to enable the ability to perform Backup and Restore operations of their own application namespace for namespace owners aka non-admin users.
+
+## Goals
+- Enable non-admin backup operation
+- Enable non-admin restore operation
+- Enable non-admin configuration of BackupStorageLocation
+
+## Non-Goals
+- Performance improvements of Backup and Restore Operations
+- Parallel Backup and Restore Operations
+
+## Use-Cases
+
+### Backup Operation
+- As a non-admin user/namespace owner with administrative priviledges for a particular namespace, the user should be able to:
+    - Create a Backup/Schedule of the namespace
+    - Update the Backup/Schedule spec of the namespace
```

Comment:
this one I am confused of what will happen. Example: if I update after Velero started (or finished) the backup process, nothing will happen, right?

Reply:
I'm not even sure what happens to Velero if you update a backup after it starts. Not sure we need to make any promises here.

Reply:
(for backups -- for schedules, those can be updated, but see above as to whether we need schedules for the first iteration)

Reply:
I guess if you update backup spec, nac could create deleteBackupRequests, mark NAB as pending, then once old backup is gone, NAB will be processing with new backup of the same name.

Reply:
In any case, updating backup spec makes sense as it just does exactly what a normal velero (admin) user modifying backup spec would do. If there's any modification that makes sense in velero, then it could be done here.